packages/fxa-settings/src/lib/metrics.ts

```diff
@@ -207,42 +207,29 @@ export async function init(enabled: boolean, flowQueryParams: QueryParams) {
   }
 }
 
+/**
+ * Initializes the user preference state for metrics.
+ * @param account - An account model containing user preferences applicable to metrics
+ */
+export function initUserPreferences(account: {
+  recoveryKey: boolean;
+  hasSecondaryVerifiedEmail: boolean;
+  totpActive: boolean;
+}) {
+  setUserPreference('account-recovery', account.recoveryKey);
+  setUserPreference('emails', account.hasSecondaryVerifiedEmail);
+  setUserPreference('two-step-authentication', account.totpActive);
+}
+
 /**
  * Often we need to log a metric event in a place where hooks are not allowed.
  * However, we do want initialize some event data through the use of hooks.
  * It also provide a couple functions that ensure the event is logged once.
  */
-export function useMetrics(): {
-  logViewEventOnce: typeof logViewEvent;
-  logPageViewEventOnce: typeof logPageViewEvent;
-} {
-  useUserPreferences();
-  const makeLogOnceFn = useRef((f: Function) => {
-    let hasLogged = false;
-
-    return (
-      ...args:
-        | Parameters<typeof logViewEvent>
-        | Parameters<typeof logPageViewEvent>
-    ) => {
-      if (!hasLogged) {
-        hasLogged = true;
-        f(...args);
-      }
-    };
-  });
-
-  const logViewEventOnce = useRef<
-    (...args: Parameters<typeof logViewEvent>) => void
-  >(makeLogOnceFn.current(logViewEvent));
-
-  const logPageViewEventOnce = useRef<
-    (...args: Parameters<typeof logPageViewEvent>) => void
-  >(makeLogOnceFn.current(logPageViewEvent));
-
+export function useMetrics() {
```

Comment:
IMO, we don't need `useMetrics` as it's another hook that doesn't seem like it really needs to be a hook. I was trying stop this PR from getting too big though, so I sort of think we should look at our hooks and figure out which ones are truly good candidates for hooks, and then clean this up in a separate task. 

If you would liked me to refactor away useMetrics, I can though.

Reply:
I agree. Next reset password ticket one of us can do this, or we could file a 1-pointer but I'm not even sure it's worth that since we only use it on two pages.